File: src/Data_Structures/Queue/queue.test.ts
import { Queue } from "./queue";

test("Should enqueue items to the left", () => {
  const queue = new Queue(4, 5);
  expect(queue.enqueue(1, 2, 3)).toBe(5);
  expect(queue.toArray()).toEqual([4, 5, 1, 2, 3]);
});

test("Should dequeue item from the right", () => {
  const queue = new Queue(1, 2, 3);
  expect(queue.dequeue()).toBe(1);
});

test("Should create a new empty queue and add items to it", () => {
  const queue = new Queue();
  expect(queue.isEmpty()).toBeTruthy();
  queue.enqueue("Learn data structures.");
  queue.enqueue("Find a job of dream.");
  queue.enqueue("Be happy!");

  expect(queue.peek()).toBe("Learn data structures.");
  expect(queue.dequeue()).toBe("Learn data structures.");
  expect(queue.peek()).toBe("Find a job of dream.");
  expect(queue.length).toBe(2);
  queue.dequeue();
  expect(queue.peek()).toBe("Be happy!");
  queue.dequeue();
  expect(queue.dequeue()).toBeUndefined();
  expect(queue.isEmpty()).toBeTruthy();
  expect(queue.length).toBe(0);
});

File: src/Data_Structures/Queue/queue_test.py
from queue import Queue


def test_enqueueItemsToLeft():
    queue = Queue(4, 5)
    assert queue.enqueue(1, 2, 3) == 5
    assert queue.to_list() == [4, 5, 1, 2, 3]


def test_dequeueItemFromRight():
    queue = Queue(1, 2, 3)
    assert queue.dequeue() == 1


def test_createEmptyQueueAndAddItems():
    queue = Queue()
    assert queue.is_empty() == True

    queue.enqueue("a")
    queue.enqueue("b")
    queue.enqueue("c")
    assert queue.peek() == "a"
    assert queue.dequeue() == "a"
    assert queue.peek() == "b"
    assert queue.get_length() == 2

    queue.dequeue()
    assert queue.peek() == "c"

    queue.dequeue()
    assert queue.dequeue() == None
    assert queue.is_empty() == True
    assert queue.get_length() == 0
